plugins/scanners/scancode/src/main/kotlin/ScanCodeResultModelMapper.kt

```diff
@@ -91,19 +91,31 @@ fun ScanCodeResult.toScanSummary(): ScanSummary {
             it.value.first()
         }
 
-        licenses.mapTo(licenseFindings) { license ->
-            // ScanCode uses its own license keys as identifiers in license expressions.
-            val spdxLicenseExpression = license.licenseExpression.mapLicense(scanCodeKeyToSpdxIdMappings)
-
-            LicenseFinding(
-                license = spdxLicenseExpression,
+        if (preferFileLicense && file is FileEntry.Version3 && file.detectedLicenseExpressionSpdx != null) {
+            licenseFindings += LicenseFinding(
+                license = file.detectedLicenseExpressionSpdx,
                 location = TextLocation(
                     path = file.path,
-                    startLine = license.startLine,
-                    endLine = license.endLine
+                    startLine = licenses.minOf { it.startLine },
+                    endLine = licenses.maxOf { it.endLine }
                 ),
-                score = license.score
+                score = licenses.map { it.score }.average().toFloat()
```

Comment:
Would you know the rational DOS use when coming up with an `avg` score? 
Also, I wonder in what way the CLI option `--license-score` influences the per file license? (and whether this should become a command line config option).

Reply:
> Would you know the rational DOS use when coming up with an `avg` score?

@Etsija and @lamppu probably know better, but the average is pretty much the only metric that somewhat makes sense and is reasonably easy to calculate. Min. does not make sense, max. does not make sense.

Of course, you could do something "crazy" like only taking the scores of those line findings into account whose licenses are also contained in the file finding, and maybe even weight the score according to the number of lines matched, but that's much more work for questionable benefit.

> I wonder in what way the CLI option --license-score influences the per file license?

Let's ask the author's: @pombredanne, does ScanCode's [`--license-score` option](https://scancode-toolkit.readthedocs.io/en/stable/cli-reference/basic-options.html#license-score-option) have an (implicit) impact on which licenses will be part of the per-file `detected_license_expression(_spdx)` fields?

Reply:
> Of course, you could do something "crazy" like only taking the scores of those line findings into account whose licenses are also contained in the file finding, and maybe even weight the score according to the number of lines matched, but that's much more work for questionable benefit.

Another alternative would be to omit `lines` and `score` entirely, with the rational that they stop being meaningful.
The reason I'm picking this up is, because (1) I suspect these values are a bit more likely to exhibit some bit of flakyness at least across scancode version (2) the result of the heuristic gets backed into the stored scan results and is thus very expensive to change.

Reply:
> Another alternative would be to omit `lines` and `score` entirely, with the rational that they stop being meaningful.

Making lines optional would be a bigger change in the data model, and it would prevent us from matching copyright to license findings, and probably have more implications.

However, I forgot that the `score`actually is optional already. Would you prefer to omit it instead of calculating the average?

> I suspect these values are a bit more likely to exhibit some bit of flakyness at least across scancode version

More likely to be flaky than what, than the individual per-line findings? As the lines and score are calculated from the per-line findings, they can only be flaky if the per-line findings themselves are already flaky.

> the result of the heuristic gets backed into the stored scan results and is thus very expensive to change.

Correct, but that's the way forward how ScanCode will operate, I guess. As [this discussion](https://github.com/nexB/scancode-toolkit/issues/3458) shows, there seem to be no endeavors to enhance the quality of the per-line findings themselves, but to achieve better quality by post-processing these findings into the per-file finding.

Reply:
> Correct, but that's the way forward how ScanCode will operate, I guess. As [this discussion](https://github.com/nexB/scancode-toolkit/issues/3458) shows, there seem to be no endeavors to enhance the quality of the per-line findings themselves, but to achieve better quality by post-processing these findings into the per-file finding.

I don't much about the processing inside of scancode, so I don't want to move this discussion off-topic,
but it's already interesting that this is actually a post-processing step of per-line findings.
If the only input of this post processing step is the a per-line findings, then it could
also / in the long run make more sense to have that postprocessing step applied by ORT (using ScanCodes logic), not by ScanCode (at least not in a single step) in order to make use of the per-line finding curations. The process could be:

1. Compute per line findings with ScanCode
2. Apply per line finding curations aka license finding curations
3. Take result from 2. as input for the post processing to compute the per-file findings using ScanCode's heuristic 
     feeding in the curated license findings.

I believe this would exhibit several advantages over the current solution. 

@pombredanne - what do you think about this idea?

Reply:
> make more sense to have that postprocessing step applied by ORT (using ScanCodes logic)

I agree, *if* that's possible, and ScanCode does not require internal data for this that's not written to the result file.

Reply:
> However, I forgot that the `score`actually is optional already. Would you prefer to omit it instead of calculating the average?

@fviernau I still need you input here. I believe that's the last thing to clarify before we could merge?

Reply:
I'd have slight tendency to omit it, but I also believe my knowledge is too limited in that area to make the call.
The reason I believe why its better to omit it, is because I believe the way the score would be defined would even harder to grasp and hardly be actionable. E.g. if an SPDX expression would consist of several 100% findings and a  few 80% findings, the final score would have a relatively low score even though some parts of the expression apply a 100%. So,e.g. filtering out on score would not be feasible anymore, because one would have two different ways the score could have been computed.

So, all in all I prefer to omit it, but would be fine if you kept it. Up to you.

Reply:
@mnonnenmacher what do you think about https://github.com/oss-review-toolkit/ort/pull/8152#discussion_r1464745604 ?